client/llb/source_test.go

```diff
@@ -0,0 +1,68 @@
+package llb
+
+import (
+	"testing"
+
+	"github.com/moby/buildkit/util/gitutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseGitRemote(t *testing.T) {
+	for _, tc := range []struct {
+		remote           string
+		expectedProtocol int
+		expectedSSHHost  string
+		expectedRemote   string
+	}{
+		{
+			remote:           "git@github.com:moby/buildkit.git",
+			expectedProtocol: gitutil.SSHProtocol,
+			expectedSSHHost:  "github.com",
+			expectedRemote:   "github.com/moby/buildkit.git",
+		},
+		{
+			remote:           "ssh://github.com:moby/buildkit.git",
+			expectedProtocol: gitutil.SSHProtocol,
+			expectedSSHHost:  "github.com",
+			expectedRemote:   "github.com/moby/buildkit.git",
+		},
+		{
+			remote:           "ssh://github.com:buildkit.git",
+			expectedProtocol: gitutil.SSHProtocol,
+			expectedSSHHost:  "github.com",
+			expectedRemote:   "github.com/buildkit.git",
+		},
+		{
+			remote:           "ssh://github.com:22/moby/buildkit.git",
+			expectedProtocol: gitutil.SSHProtocol,
+			expectedSSHHost:  "github.com:22",
+			expectedRemote:   "github.com:22/moby/buildkit.git",
+		},
+		{
+			remote:           "ssh://github.com:22:moby/buildkit.git",
+			expectedProtocol: gitutil.SSHProtocol,
+			expectedSSHHost:  "github.com",
```

Comment:
Expected host here should be `github.com:22:moby` if we want to follow the same behavior as git:

```
$ git clone ssh://git@github.com:moby:22/buildkit.git
Cloning into 'buildkit'...
ssh: Could not resolve hostname github.com:moby:22: Name or service not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```